tests/python/rest_api/test_tasks.py

```diff
@@ -1012,6 +1012,71 @@ def test_datumaro_export_without_annotations_includes_image_info(
                     if "size" in related_image:
                         assert tuple(related_image["size"]) > (0, 0)
 
+    @pytest.mark.usefixtures("restore_db_per_function")
+    @pytest.mark.usefixtures("restore_redis_ondisk_per_function")
+    @pytest.mark.parametrize(
+        "format_name, num_frames, frame_step, expected_frames",
+        [
+            ("Datumaro 1.0", 100, 5, 20),
+            ("COCO 1.0", 100, 5, 20),
+            ("CVAT for video 1.1", 100, 5, 20),  # no remainder
+            ("CVAT for video 1.1", 97, 5, 20),  # prime
+            ("CVAT for video 1.1", 97, 2, 49),
+            ("CVAT for video 1.1", 100, 3, 34),  # three
+            # we assert that expected frames are ceil(frames / step)
+        ],
+    )
+    def test_export_with_non_default_frame_step(
+        self,
+        # fixtures
+        tmp_path: Path,
+        admin_user: str,
+        # params
+        format_name: str,
+        num_frames: int,
+        frame_step: int,
+        expected_frames: int,
+    ):
+        # parameter validation
+        assert expected_frames == math.ceil(num_frames / frame_step), "Test params are wrong"
+
+        spec = {
+            "name": f"test_video_frames_in_{format_name}_after_export",
+            "labels": [{"name": "goofy ahh car"}],
+        }
+
+        data = {
+            "image_quality": 70,
+            "client_files": [generate_video_file(num_frames)],
+            "frame_filter": f"step={frame_step}",
+        }
+
+        # create a task and get its instance
+        (task_id, _) = create_task(admin_user, spec, data)
+        with make_sdk_client(admin_user) as client:
+            task_obj: Task = client.tasks.retrieve(task_id)
+
+        # export the video
+        dataset_file = tmp_path / "dataset.zip"
+        task_obj.export_dataset(format_name, dataset_file, include_images=True)
+
+        def get_png_index(zinfo: zipfile.ZipInfo) -> int:
+            name = PurePosixPath(zinfo.filename)
+            if name.suffix.lower() != ".png":  # png is usually for video
```

Comment:
```suggestion
```
I don't think it makes sense to add some branching here in the test. There is only 1 expected file format.